lib/pbench/server/api/resources/query_apis/query_month_indices.py

```diff
@@ -0,0 +1,101 @@
+from flask import jsonify
+from flask_restful import Resource, abort
+import requests
+
+from pbench.server.api.resources.query_apis import get_es_url, get_index_prefix
+
+
+class QueryMonthIndices(Resource):
+    """
+    Abstracted Pbench API to get date-bounded controller data.
+    """
+
+    def __init__(self, config, logger):
+        self.logger = logger
+        self.es_url = get_es_url(config)
+        self.prefix = get_index_prefix(config)
+
+    def get(self):
+        """
+        GET to detect the month suffixes existing for run data indices.
+
+        Required headers include
+
+        Content-Type:   application/json
+        Accept:         application/json
+
+        The return payload is a list of "YYYY-mm" date strings corresponding
+        to the months in which tarballs were indexed into the appropriate
+        `run-data` index. (E.g., `drb.v6.run-data.2020-11`). Note that this
+        list is in DESCENDING order, so the earliest date is last.
+
+        NOTE: No authorization or input payload is required for this API.
+
+        NOTE: This is the format currently constructed by the Pbench
+        dashboard `src/model/datastore.js` fetchMonthIndices method, which
+        becomes part of the Redux state.
+
+        [
+            "2020-04",
+            "2020-11",
+            "2020-12"
+        ]
+        """
+
+        self.logger.info(
+            "QueryMonthIndices GET for prefix {}", self.prefix,
+        )
+
+        uri = f"{self.es_url}/_aliases"
+        try:
+            # query Elasticsearch
+            es_response = requests.get(uri, headers={"Accept": "application/json"})
+            es_response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            self.logger.exception("HTTP error {} from Elasticsearch post request", e)
+            abort(500, message=f"HTTP error {e} from Elasticsearch")
```

Comment:
Not sure we should return a `500` error here.  This seems like a normal operation of this server.  Perhaps returning `502` instead?  Not sure we should pass along the exception, either.

Reply:
It's an interesting question; it really depends on what Elasticsearch is complaining about. We could just pass on whatever it said; I'm not sure if there's a useful payload on the HTTPError exception, and the basic niceness of `raise_for_status()` is not having to decode the HTTP error codes.

This is, in any case, just a copy-and-paste from our other ES interface code, so solving this in general across all of them probably makes sense. I can try to do that here, or we can deal with it separately.

Reply:
Separately.

Let's just return an "INTERNAL ERROR" for now, we have the full exception in the logs.  It is unclear what the dashboard would do with the full message any ways.

Reply:
The Elasticsearch passthrough actually returns `es_response.status_code` ... that makes perfect sense for the passthrough, since the caller is directly responsible for any syntactic errors in the request payload. I added the same logic to the GraphQL passthrough as it makes equal sense there.

It's less clear for these, as Pbench is responsible for any syntactic errors, and we shouldn't be letting that happen. I'll go with 502 and `INTERNAL ERROR` for now, and make sure we're consistent.

Reply:
The dashboard contains a component to handle errors with appropriate messages. This component can be made use of. https://github.com/distributed-system-analysis/pbench-dashboard/blob/master/src/pages/Exception/triggerException.js

Reply:
However currently, there is no way to include "messages" to be displayed along with this component. Just the http_status_code.